src/GraphQL/Federation/FederationHelper.cs

```diff
@@ -246,7 +246,7 @@ void RegisterDirective(string directiveName,
                 return;
             var directive = new Directive(link.NameForDirective(directiveName))
             {
-                Arguments = arguments != null ? new QueryArguments(arguments) : null,
+                Arguments = arguments != null ? [.. arguments] : null,
```

Comment:
Again, performance is unclear here.  Does it call the constructor that accepts an enumerable or call Add a bunch of times?